File: java/Cord.java
package path_find;

public class Cord {
    int x;
    int y;
    Cord prevCord;

    Cord(int x, int y, Cord cord) {
        this.x = x;
        this.y = y;
        this.prevCord = cord;
    }

    Cord(int x, int y) {
        this.x = x;
        this.y = y;
    }

    Cord(int x, int y, int prev_x, int prev_y) {
        this.x = x;
        this.y = y;
        this.prevCord = new Cord(prev_x, prev_y);
    }
}

File: c#/Cord.cs
namespace path_find;

public class Cord
{
    public int x;
    public int y;

    public Cord? prevCord;

    public Cord(int x, int y, Cord? prevCord)
    {
        this.x = x;
        this.y = y;
        this.prevCord = prevCord;
    }
}